xbmc/Application.cpp

```diff
@@ -4211,9 +4213,9 @@ bool CApplication::OnMessage(CGUIMessage& message)
         CGUIMessage msg(GUI_MSG_PLAYLISTPLAYER_CHANGED, 0, 0, g_playlistPlayer.GetCurrentPlaylist(), param, item);
         g_windowManager.SendThreadMessage(msg);
         g_playlistPlayer.SetCurrentSong(m_nextPlaylistItem);
-        *m_itemCurrentFile = *item;
+        m_itemCurrentFile = item;
```

Comment:
and? wasn't the original purpose to "deep copy" the item into m_itemCurrentFile? now you're just copying the (shared) reference. I'd be more comfortable if you changed this to m_itemCurrentFile.reset(new CFileItem(*item));